### ISSUE TITLE
`just clippy` does not catch warnings produced by `just dnssec-openssl`

### ISSUE THREAD

**Issue by japaric** [hickory-dns/hickory-dns#2212]
**Describe the bug**
what the title says

**To Reproduce**

``` console
$ # from the root of the repository
$ just dnssec-openssl
cargo ws exec --ignore=\{async-std-resolver,hickory-compatibility\} cargo  check --all-targets --benches --examples --bins --tests --features=dnssec-openssl
(..)
warning: unused import: `time::Duration`
 --> tests/integration-tests/tests/client_tests.rs:9:5
  |
9 | use time::Duration;
  |     ^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `time::Duration`
 --> tests/integration-tests/tests/client_future_tests.rs:9:5
  |
9 | use time::Duration;
  |     ^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `hickory_proto::rr::Record`
  --> tests/integration-tests/tests/client_tests.rs:27:5
   |
27 | use hickory_proto::rr::Record;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused imports: `Record`, `dnssec::SigSigner`
  --> tests/integration-tests/tests/client_future_tests.rs:32:25
   |
32 | use hickory_proto::rr::{dnssec::SigSigner, Record};
   |                         ^^^^^^^^^^^^^^^^^  ^^^^^^

warning: unused imports: `DnsExchangeBackground`, `DnsMultiplexer`
  --> tests/integration-tests/tests/client_future_tests.rs:34:27
   |
34 | use hickory_proto::xfer::{DnsExchangeBackground, DnsMultiplexer};
   |                           ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^
(..)
```

``` console
$ just clippy && echo OK
(..)
OK
```

**Expected behavior**
`just clippy` should catch the warnings shown when one runs `just dnssec-openssl` and similar tasks.

**System:**
 - OS: Arch Linux
 - Architecture: x86_64
 - Version N/A
 - rustc version: 1.76.0

**Version:**
Crate: whole repository
Version: 97e1f434

**Additional context**

Seems like `just clippy` only test two combinations of Cargo features: all features disabled; and all features enabled. 

To test the combinations of other Cargo features a tool like [`cargo-hack`](https://crates.io/crates/cargo-hack) could be used. I know the rustls project uses it to test combinations of Cargo features but I have not used it myself.

**Comment by djc** [comment]
I'd be open to a PR that introduces the use of cargo-hack similar to what we have in the rustls repo. Might want a separate job for it since it could be pretty slow?

**Comment by bluejekyll** [comment]
I keep hesitating on the removal of OpenSSL because of some of the compliance it has that we're still waiting on for Rustle. I think we should just clean it up instead of fixing up testing related to it, but it wouldn't be hard to fix this.

Today we only test these feature flags directly with clippy:

```
clippy:
    find {{justfile_directory()}} -name '*.rs' -exec touch {} \;
    just clippy-inner --no-default-features
    just clippy-inner
    just clippy-inner --all-features
```

We could add this after `--all-features`, `just clippy-inner --features=dnssec-openssl`. OpenSSL generally doesn't get caught, because where they conflict, Rustls takes precedent.

**Comment by djc** [comment]
> I keep hesitating on the removal of OpenSSL because of some of the compliance it has that we're still waiting on for Rustle. I think we should just clean it up instead of fixing up testing related to it, but it wouldn't be hard to fix this.

What kind of compliance are we waiting for? I think Rustls is in a pretty good place today.

**Comment by bluejekyll** [comment]
> What kind of compliance are we waiting for?

Primarily, FIPS compliance, does this announcement mean that Rustls is FIPS compliance now? https://www.memorysafety.org/blog/rustls-with-aws-crypto-back-end-and-fips/

**Comment by djc** [comment]
Yes, Rustls from 0.22 onwards supports the aws-lc-rs crypto provider which has a FIPS mode available (on Linux only).

**Comment by mokeyish** [comment]
> Yes, Rustls from 0.22 onwards supports the aws-lc-rs crypto provider which has a FIPS mode available (on Linux only).

`aws-lc-rs` compiles failed on target arm-unknown-linux-musleabi

![image](https://github.com/user-attachments/assets/ed4c2ff2-c585-4d8a-b103-2e3fa0caab47)

**Comment by djc** [comment]
Support for OpenSSL was removed:

- #2656
